demo/fake-api.js

```diff
@@ -0,0 +1,614 @@
+          },
+          ManyRunningProcesses: {
+            score: 0.20309488837692125,
+            name: "Competing Processes",
+            topic: "processes",
+            description:
+              "Summary:\n1 processes were running while this Flight Recording was made.\n\nExplanation:\nAt 5/5/23, 5:17:27.180 PM, a total of 1 other processes were running on the host machine that this Flight Recording was made on.\n\nSolution:\nIf this is a server environment, it may be good to only run other critical processes on that machine.",
+          },
+          StackdepthSetting: {
+            score: 25.0,
+            name: "Stackdepth Setting",
+            topic: "jvm_information",
+            description:
+              "Summary:\nSome stack traces were truncated in this recording.\n\nExplanation:\nThe Flight Recorder is configured with a maximum captured stack depth of 64. 3.11 % of all traces were larger than this option, and were therefore truncated. If more detailed traces are required, increase the \u0027\u0027-XX:FlightRecorderOptions\u003dstackdepth\u003d\u003cvalue\u003e\u0027\u0027 value.\nEvents of the following types have truncated stack traces: org.openjdk.jmc.flightrecorder.rules.jdk.general.StackDepthSettingRule$StackDepthTruncationData@21e159e2,org.openjdk.jmc.flightrecorder.rules.jdk.general.StackDepthSettingRule$StackDepthTruncationData@174930bc,org.openjdk.jmc.flightrecorder.rules.jdk.general.StackDepthSettingRule$StackDepthTruncationData@4f5d6223",
+          },
+          PasswordsInArguments: {
+            score: 0.0,
+            name: "Passwords in Java Arguments",
+            topic: "jvm_information",
+            description:
+              "Summary:\nThe recording does not seem to contain passwords in the application arguments.",
+          },
+        }
+      );
+    });
+    this.get("/api/v1/targets/:targetId/recordingOptions", () => []);
+    this.get("/api/v1/targets/:targetId/events", () => [
+      {
+        category: ["GC", "Java Virtual Machine"],
+        name: "GC Heap Configuration",
+        typeId: "jdk.GCHeapConfiguration",
+        description: "The configuration of the garbage collected heap",
+      },
+    ]);
+    this.get("/api/v1/targets/:targetId/templates", () => [
+      {
+        name: "Demo Template",
+        provider: "Demo",
+        type: "TARGET",
+        description: "This is not a real event template, but it is here!",
+      },
+    ]);
+    this.get("/api/v2/probes", () => []);
+
+    this.post("/api/v2/rules", (schema, request) => {
+      let attrs = JSON.parse(request.requestBody);
+      return {
+        data: {
+          result: schema.rules.create(attrs),
+        },
+      };
+    });
+    this.get("/api/v2/rules", (schema) => ({
+      data: { result: schema.rules.all().models },
+    }));
+    this.get("/api/v2.2/credentials", () => ({ data: { result: [] } }));
+
+    this.post("/api/v2.2/graphql", (schema, request) => {
```

Comment:
Seems like JFR Metrics is not mocked out? The recording is created as expected but the card does not see it.

![Screenshot from 2023-05-09 14-36-42](https://github.com/cryostatio/cryostatio.github.io/assets/68053619/15c09f05-74d6-4e94-b966-097475ad3ec1)

Reply:
Yea, not everything is mocked out here. I didn't intend for this to be an entirely full-featured mock demo, so some views like this one and the All Targets view are just unimplemented. I put in the `400 Bad Reqest, unsupported in demo` to generally cover those cases of unimplemented views. Otherwise, trying to keep this stubbed out demo up-to-date with actual developments in the backend and frontend will become a large maintenance burden of its own.

Reply:
> Would it be better (if possible) generate these demo assets + stubs into a special branch in the cryostat-web (or another repo called web-preview) that is setup with GitHubPages? Then, have some CI to automate that task? Here, we can just reference that public github page URL.

I am thinking if this is possible to set up this demo/preview into `cryostat-web`, we can start to fully mock and have it as a preview for our latest changes. Similar to https://github.com/patternfly/patternfly-react.

Reply:
Yep, I was writing another comment on the main PR thread below saying basically the same thing :-)